lightning/backend/tidb.go

```diff
@@ -401,7 +459,44 @@ func (be *tidbBackend) FetchRemoteTableModels(schemaName string) (tables []*mode
 			})
 			curColOffset++
 		}
-		return rows.Err()
+		if rows.Err() != nil {
+			return rows.Err()
+		}
+		// init auto id column for each table
+		for _, tbl := range tables {
+			tblName := common.UniqueTable(schemaName, tbl.Name.O)
+			rows, e = tx.Query(fmt.Sprintf("SHOW TABLE %s next_row_id", tblName))
+			if e != nil {
+				return e
+			}
+			for rows.Next() {
+				var (
+					dbName, tblName, columnName, idType string
+					nextID                              int64
+				)
+				if e := rows.Scan(&dbName, &tblName, &columnName, &nextID, &idType); e != nil {
+					_ = rows.Close()
+					return e
+				}
+				for _, col := range tbl.Columns {
+					if col.Name.O == columnName {
+						switch idType {
+						case "AUTO_INCREMENT":
+							col.Flag |= mysql.AutoIncrementFlag
+						case "AUTO_RANDOM":
+							col.Flag |= mysql.PriKeyFlag
+							// TODO: we don't know the actual auto random bits here, so we use the default value.
+							tbl.AutoRandomBits = autoid.DefaultAutoRandomBits
```

Comment:
```sql
SELECT auto_increment, tidb_row_id_sharding_info FROM information_schema.tables;
```

if there is an auto_increment_id, the `auto_increment` field is not null. if there is auto_random_id, the `tidb_row_id_sharding_info` contains the string like `"PK_AUTO_RANDOM_BITS=3"` (otherwise `"NOT_SHARDED"`).